app/models/solidus_bolt/gateway.rb

```diff
@@ -59,12 +80,19 @@ def purchase(_float_amount, _response_code, _gateway_options)
 
     private
 
-    def can_capture?(reference, payment_method)
-      transaction_detail = SolidusBolt::Transactions::DetailService.call(
+    def fetch_transaction_detail(reference, payment_method)
+      SolidusBolt::Transactions::DetailService.call(
         transaction_reference: reference,
         payment_method: payment_method
       )
+    end
 
+    def can_void?(payment)
+      payment.state != 'void'
+    end
```

Comment:
I know it's on the card requirement, but can_void? is already provided by the [state_machine](https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/core/lib/spree/core/state_machines/payment.rb#L41-L43), maybe we don't have to implement that method. WDYT?

Reply:
removed